core/chains/evm/client/chain_client.go

```diff
@@ -0,0 +1,243 @@
+	return c.multiNodeClient.SelectNode().RPCClient().PendingSequenceAt(ctx, addr)
+}
+
+func (c *chainClient) SendEmptyTransaction(
+	ctx context.Context,
+	newTxAttempt func(seq evmtypes.Nonce, feeLimit uint32, fee *assets.Wei, fromAddress common.Address) (attempt any, err error),
+	seq evmtypes.Nonce,
+	gasLimit uint32,
+	fee *assets.Wei,
+	fromAddress common.Address,
+) (txhash string, err error) {
+	return c.multiNodeClient.SelectNode().RPCClient().SendEmptyTransaction(ctx, newTxAttempt, seq, gasLimit, fee, fromAddress)
+}
+
+func (c *chainClient) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	main := c.multiNodeClient.SelectNode()
+	var all []clienttypes.Node[*big.Int, common.Hash, *evmtypes.Head, ethereum.Subscription, RPCClient]
+	all = append(all, c.nodes...)
+	all = append(all, c.sendonlys...)
+	for _, n := range all {
+		if n == main {
+			// main node is used at the end for the return value
+			continue
+		}
+		// Parallel send to all other nodes with ignored return value
+		// Async - we do not want to block the main thread with secondary nodes
+		// in case they are unreliable/slow.
+		// It is purely a "best effort" send.
+		// Resource is not unbounded because the default context has a timeout.
+		ok := c.IfNotStopped(func() {
+			// Must wrap inside IfNotStopped to avoid waitgroup racing with Close
+			c.wg.Add(1)
+			go func(n clienttypes.Node[*big.Int, common.Hash, *evmtypes.Head, ethereum.Subscription, RPCClient]) {
+				defer c.wg.Done()
+
+				sendCtx, cancel := c.chStop.CtxCancel(ContextWithDefaultTimeout())
+				defer cancel()
+				err, _ := n.RPCClient().SendTransactionReturnCode(sendCtx, tx)
+				c.logger.Debugw("Sendonly node sent transaction", "name", n.String(), "tx", tx, "err", err)
+				if err == clienttypes.TransactionAlreadyKnown || err == clienttypes.Successful {
+					// Nonce too low or transaction known errors are expected since
+					// the primary SendTransaction may well have succeeded already
+					return
+				}
+				c.logger.Warnw("Eth client returned error", "name", n.String(), "err", err, "tx", tx)
+			}(n)
+		})
+		if !ok {
+			c.logger.Debug("Cannot send transaction on sendonly node; pool is stopped", "node", n.String())
+		}
+	}
+
+	return main.RPCClient().SendTransaction(ctx, tx)
+}
+
+func (c *chainClient) SendTransactionReturnCode(
+	ctx context.Context,
+	tx *types.Transaction,
+) (clienttypes.SendTxReturnCode, error) {
+	return c.multiNodeClient.SelectNode().RPCClient().SendTransactionReturnCode(ctx, tx)
+}
+
+func (c *chainClient) SequenceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (evmtypes.Nonce, error) {
+	return c.multiNodeClient.SelectNode().RPCClient().SequenceAt(ctx, account, blockNumber)
+}
+
+func (c *chainClient) SimulateTransaction(ctx context.Context, tx *types.Transaction) error {
+	return c.multiNodeClient.SelectNode().RPCClient().SimulateTransaction(ctx, tx)
+}
```

Comment:
Are these go-ethereum methods? Haven't seen them before

Reply:
These are from the generalized RPC Client interface that we had created initially to be applicable to any chain. SequenceAt is basically just an alias for [NonceAt](https://github.com/smartcontractkit/chainlink/blob/d1e49c7623d46eaf8473a081fb62d0d47c92089f/core/chains/evm/client/rpc_client.go#L527-L557) in this case. SimulateTransaction was a Solana-specific method that we decided could be applicable to Ethereum so we left it in the generalized interface. It's currently not implemented [for EVM](https://github.com/smartcontractkit/chainlink/blob/d1e49c7623d46eaf8473a081fb62d0d47c92089f/core/chains/evm/client/rpc_client.go#L480-L483) though.